src/terrain_3d_material.cpp

```diff
@@ -231,65 +236,112 @@ void Terrain3DMaterial::_update_texture_arrays(const Array &p_args) {
 		}
 	} else {
 		set_show_checkered(false);
-		LOG(DEBUG, "Texture count >0, disabling checkered view");
+		LOG(DEBUG, "Texture count >0: ", _texture_count, ", disabling checkered view");
 	}
 }
 
 void Terrain3DMaterial::_update_shader() {
+	if (!_initialized) {
+		return;
+	}
 	LOG(INFO, "Updating shader");
 	if (_shader_override_enabled && _shader_override.is_valid()) {
+		if (_shader_override->get_code().is_empty()) {
+			String code = _generate_shader_code();
+			_shader_override->set_code(code);
+		}
+		if (!_shader_override->is_connected("changed", Callable(this, "_update_shader"))) {
+			LOG(DEBUG, "Connecting changed signal to _update_shader()");
+			_shader_override->connect("changed", Callable(this, "_update_shader"));
+		}
 		RS->material_set_shader(_material, _shader_override->get_rid());
+		LOG(DEBUG, "Mat rid: ", _material, ", _shader_override rid: ", _shader_override->get_rid());
 	} else {
 		RS->shader_set_code(_shader, _generate_shader_code());
 		RS->material_set_shader(_material, _shader);
+		LOG(DEBUG, "Mat rid: ", _material, ", _shader rid: ", _shader);
+	}
+	// Update custom shader params in RenderingServer
+	// Populate _active_params
+	_get_property_list(&(List<PropertyInfo>()));
```

Comment:
This results in a compile error on my machine

`src/terrain_3d_material.cpp:266:21: error: taking the address of a temporary object of type 'List<PropertyInfo>' [-Waddress-of-temporary]`

This works: 

```suggestion
	{
		auto temp_list = List<PropertyInfo>();
		_get_property_list(&temp_list);
	}
```
```c++

Reply:
Thanks for testing. Godot devs don't like auto, but I can put in a typed temp var. What compiler are you using? Apple clang?

Reply:
Yeah, Apple clang 15. Just the default setting from the godot_cpp SConstruct